rust-runtime/aws-smithy-http/src/header.rs

```diff
@@ -132,8 +130,8 @@ fn read_many<T>(
 /// Read exactly one or none from a headers iterator
 ///
 /// This function does not perform comma splitting like `read_many`
-pub fn one_or_none<T: FromStr>(
-    mut values: ValueIter<'_, HeaderValue>,
+pub fn one_or_none<'a, T: FromStr>(
```

Comment:
(can't leave a comment on line 144). Shouldn't we now no longer need the `std::str::from_utf8` call?

Reply:
In general this file contains several `.as_bytes` calls, and code that works with `&[u8]` when I think it could directly work with `&str` all the way through?

Reply:
Oh, line 144 looks pretty bad. I'll fix that one at the very least. It'll take a good bit of effort to fix the rest of the file though, it looks like.

Touched up in 85af0354cf2a91e4a065a5232b9357ab40e6257f.